src/targets/gpu/convolution.cpp

```diff
@@ -9,15 +9,38 @@ namespace gpu {
 shape miopen_convolution::compute_shape(const std::vector<shape>& inputs) const
 {
     check_shapes{inputs, *this}.has(4).standard();
-    return op.compute_shape({inputs.at(0), inputs.at(1)});
+    std::vector<shape> conv_inputs(inputs.begin(), inputs.begin() + 2);
+    check_shapes{conv_inputs, *this}.max_ndims(5);
+    return op.compute_shape(conv_inputs);
 }
+
+inline void recompute_shape_to_2d(shape& input)
+{
+    auto dims = input.lens();
+
+    if(dims.size() == 3)
+    {
+        std::vector<size_t> new_dims = dims;
+        new_dims.insert(new_dims.begin() + 2, 1);
+        input = shape{input.type(), new_dims};
+    }
+}
+
 argument miopen_convolution::compute(context& ctx,
                                      const shape& output_shape,
                                      const std::vector<argument>& args) const
 {
-    auto x_desc = make_tensor(args[0].get_shape());
-    auto w_desc = make_tensor(args[1].get_shape());
-    auto y_desc = make_tensor(output_shape);
+    shape x_shape = args[0].get_shape();
+    shape w_shape = args[1].get_shape();
+    shape y_shape = output_shape;
+
+    recompute_shape_to_2d(x_shape);
+    recompute_shape_to_2d(w_shape);
+    recompute_shape_to_2d(y_shape);
+
+    auto x_desc = make_tensor(x_shape);
```

Comment:
These can be written as `auto x_desc = make_tensor(recompute_shape_to_2d(x_shape));`

Reply:
fixed